### ISSUE TITLE
[Metricbeat] system.fsstat.total_files hits long MAX

### ISSUE THREAD

**Issue by johtani** [elastic/beats#5386]
- Version: 6.0.0-rc1
- Operating System: Mac OSX High Sierra

I got an error from Metricbeat on Mac OSX High Sierra.
```
2017-10-11T13:37:50+09:00 WARN Can not index event (status=400): {"type":"mapper_parsing_exception","reason":"failed to parse [system.fsstat.total_files]","caused_by":{"type":"json_parse_exception","reason":"Numeric value (9223372036854776501) out of range of long (-9223372036854775808 - 9223372036854775807)\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@254b5ad6; line: 1, column: 105]"}}
```

So, I update the index template from `long` to `double`, then it works.
I have 1TB one disk and `df -h` output is here.

```
$ df -h
Filesystem                                                                             Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1                                                                          931Gi  303Gi  621Gi    33% 3067116 9223372036851708691    0%   /
devfs                                                                                 201Ki  201Ki    0Bi   100%     696                   0  100%   /dev
/dev/disk1s4                                                                          931Gi  6.0Gi  621Gi     1%       6 9223372036854775801    0%   /private/var/vm
map -hosts                                                                              0Bi    0Bi    0Bi   100%       0                   0  100%   /net
map auto_home                                                                           0Bi    0Bi    0Bi   100%       0                   0  100%   /home
/Users/johtani/Library/PreferencePanes/Choosy.prefPane/Contents/Resources/Choosy.app  931Gi  309Gi  620Gi    34% 3125960 9223372036851649847    0%   /private/var/folders/kx/k58mcpyj2r7dfffbc43dmg100000gn/T/AppTranslocation/96DF7D79-21CA-417A-9463-95200491064E
```

And `mount` command output is here
```
/dev/disk1s1 on / (apfs, local, journaled)
devfs on /dev (devfs, local, nobrowse)
/dev/disk1s4 on /private/var/vm (apfs, local, noexec, journaled, noatime, nobrowse)
map -hosts on /net (autofs, nosuid, automounted, nobrowse)
map auto_home on /home (autofs, automounted, nobrowse)
/Users/johtani/Library/PreferencePanes/Choosy.prefPane/Contents/Resources/Choosy.app on /private/var/folders/kx/k58mcpyj2r7dfffbc43dmg100000gn/T/AppTranslocation/694E1B58-BA02-46EC-BCB9-A6C30226415D (nullfs, local, nodev, nosuid, read-only, nobrowse, mounted by johtani)
```

**Comment by andrewkroh** [comment]
It does look like your filesystem was updated from HFS to APFS in the High Sierra upgrade. 

`system.fsstat.total_files` is summed up across all filesystems. It's final value must be greater a `long`.

**Comment by liketic** [comment]
@andrewkroh Should we use `float` ? or `bigint`: https://golang.org/pkg/math/big/#NewInt ?

**Comment by ruflin** [comment]
@liketic I think the problem is not on the Golang side but elasticsearch.

@johtani I wonder if the value that is provided by the OS is actually correct? Do you really have that many files or is that a High Sierra issue?

**Comment by johtani** [comment]
@ruflin I think the value is correct each single drive.
But I installed [choosy](https://www.choosyosx.com) and it appeared in `df -h` result, so that is affected the wrong number... Maybe because of choosy, the number has doubled.
However, if I had external usb disks or something, I can have the many files.

Unfortunately, I'm not sure whether this is a High Sierra issue or has other macOS...

**Comment by ruflin** [comment]
@johtani Could you run metricbeat with `filesystem` metricset enabled and put the output here? I would be interested to see which of these volumes has so many files. As the above is happening on a user machine, I'm still suprised by the very high number of files.

**Comment by johtani** [comment]
```
{
  "_index": "metricbeat-6.0.0-2017.12.08",
  "_type": "doc",
  "_id": "jkkCNGAB8avlomecNCU7",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2017-12-08T02:44:07.787Z",
    "metricset": {
      "rtt": 81,
      "module": "system",
      "name": "filesystem"
    },
    "beat": {
      "name": "johtani-mac-15.local",
      "hostname": "johtani-mac-15.local",
      "version": "6.0.0"
    },
    "system": {
      "filesystem": {
        "mount_point": "/",
        "free_files": 9223372036851263000,
        "used": {
          "pct": 0.3539,
          "bytes": 353749094400
        },
        "type": "apfs",
        "total": 999590961152,
        "free": 645841866752,
        "device_name": "/dev/disk1s1",
        "available": 638665568256,
        "files": 9223372036854776000
      }
    }
  },
  "fields": {
    "@timestamp": [
      "2017-12-08T02:44:07.787Z"
    ]
  },
  "sort": [
    1512701047787
  ]
}
```

**Comment by ruflin** [comment]
@johtani Sorry for the late reply. Didn't really have time for this and don't think I will get to this very soon. I wonder if @andrewkroh has some more thoughts on this and how we should move forward here?

**Comment by johtani** [comment]
I think it's OK. I have never seen others who have same issue.

**Comment by andig** [comment]
I actually have the same problem. Since the high sierra APFS upgrade the ifree count is huge:

	Filesystem    512-blocks      Used Available Capacity iused               ifree %iused  Mounted on
	/dev/disk1s1   489620264 407894784  74133064    85% 2488703 9223372036852287104    0%   /
	/dev/disk1s4   489620264   6291744  74133064     8%       3 9223372036854775804    0%   /private/var/vm

**Comment by tsg** [comment]
I can confirm this both with `df`:

```
Filesystem     Size   Used  Avail Capacity iused               ifree %iused  Mounted on
/dev/disk1s1  466Gi  134Gi  328Gi    30% 1783397 9223372036852992410    0%   /
```

and Metricbeat:

```
{
  "@timestamp": "2018-02-05T14:54:32.511Z",
  "@metadata": {
    "beat": "metricbeat",
    "type": "doc",
    "version": "6.2.0"
  },
  "metricset": {
    "name": "filesystem",
    "module": "system",
    "rtt": 2498
  },
  "system": {
    "filesystem": {
      "used": {
        "pct": 0.2886,
        "bytes": 144301969408
      },
      "device_name": "/dev/disk1s1",
      "free": 355661205504,
      "available": 351749705728,
      "files": 9223372036854775807,
      "free_files": 9223372036852992390,
      "type": "apfs",
      "total": 499963174912,
      "mount_point": "/"
    }
  },
  "beat": {
    "name": "where-is-my-esc-key.local",
    "hostname": "where-is-my-esc-key.local",
    "version": "6.2.0"
  }
}```

Since the `df` output matches Metricbeat, I suspect this is an OS breaking change or bug, but didn't yet dig into it.

**Comment by simitt** [comment]
I have the same issue for `system.fsstat.total_files` on High Sierra.

**Comment by wangzhongkuo** [comment]
@johtani 
How to update the index template from long to double？
Thank you.

**Comment by jsoriano** [comment]
We should probably use double in these gauge fields that can possibly have very big values, to be solved as part of #9271.
